src/pages/storage/forms/DuplicateVolumeForm.tsx

```diff
@@ -0,0 +1,279 @@
+import ItemName from "components/ItemName";
+import StoragePoolSelector from "../StoragePoolSelector";
+import { checkDuplicateName } from "util/helpers";
+
+interface Props {
+  volume: LxdStorageVolume;
+  close: () => void;
+}
+
+export interface LxdStorageVolumeDuplicate {
+  name: string;
+  project: string;
+  pool: string;
+  member: string;
+  volumeOnly: boolean;
+}
+
+const DuplicateVolumeForm: FC<Props> = ({ volume, close }) => {
+  const toastNotify = useToastNotification();
+  const { data: settings } = useSettings();
+  const isClustered = isClusteredServer(settings);
+  const controllerState = useState<AbortController | null>(null);
+  const navigate = useNavigate();
+  const eventQueue = useEventQueue();
+  const { hasStorageVolumesAll } = useSupportedFeatures();
+
+  const { data: projects = [], isLoading: projectsLoading } = useQuery({
+    queryKey: [queryKeys.projects],
+    queryFn: fetchProjects,
+  });
+
+  const { data: volumes = [], isLoading: volumesLoading } = useQuery({
+    queryKey: [queryKeys.customVolumes, volume.project],
+    queryFn: () => loadCustomVolumes(volume.project, hasStorageVolumesAll),
+  });
+
+  const notifySuccess = (volumeName: string, project: string, pool: string) => {
+    const message = (
+      <>
+        Created volume <strong>{volumeName}</strong>.
+      </>
+    );
+
+    const redirectUrl = `/ui/project/${project}/storage/pool/${pool}/volumes/custom/${volumeName}/configuration`;
+    const actions = [
+      {
+        label: "Configure",
+        onClick: () => navigate(redirectUrl),
+      },
+    ];
+
+    toastNotify.success(message, actions);
+  };
+
+  const getDuplicatedVolumeName = (volume: LxdStorageVolume): string => {
+    const volumeNames = volumes.map((volume) => volume.name);
+    const newVolumeName = volume.name + "-duplicate";
+
+    if (volumeNames.includes(newVolumeName)) {
+      let count = 1;
+      while (volumeNames.includes(`${newVolumeName}-${count}`)) {
+        count++;
+      }
+      return `${newVolumeName}-${count}`;
+    }
+
+    return newVolumeName;
+  };
+
+  // this validation checks given changes in any one of the fields (name, project, pool)
+  // if the volume name already exists in the target project and storage pool
+  const validationSchema = Yup.object()
+    .shape({
+      name: Yup.string().required("Volume name is required"),
+      project: Yup.string().required(),
+      pool: Yup.string().required(),
+    })
+    .test("deduplicate", "", async function (values) {
+      const { name, project, pool } = values;
+      const notFound = await checkDuplicateName(
+        name,
+        project || "",
+        controllerState,
+        `storage-pools/${pool}/volumes/custom`,
+      );
+
+      if (notFound) {
+        return true;
+      }
+
+      return this.createError({
+        path: "name",
+        message:
+          "A volume with this name already exist in the target project and storage pool",
+      });
```

Comment:
Usually we do this:
```
      .test("deduplicate", "A profile with this name already exists", (value) =>
        checkDuplicateName(value, project, controllerState, "profiles"),
      )
```
Not sure why this looks more complex?

Reply:
the example you've shown checks duplicate by making api call to a fixed project. In the current scenario, the target project and storage pool may change depending on the form inputs. This means we need to check duplication against the adjusted `project` and `pool` formik fields.